nova_vm/src/ecmascript/syntax_directed_operations/scope_analysis.rs

```diff
@@ -1038,11 +1057,17 @@ impl<'a> TopLevelLexicallyDeclaredNames<'a> for Statement<'_> {
 /// The syntax-directed operation TopLevelLexicallyScopedDeclarations takes no
 /// arguments and returns a List of Parse Nodes.
 trait TopLevelLexicallyScopedDeclarations<'a> {
-    fn top_level_lexically_scoped_declarations<F: FnMut(&'a Declaration<'a>)>(&'a self, f: &mut F);
+    fn top_level_lexically_scoped_declarations<F: FnMut(LexicallyScopedDeclaration<'a>)>(
+        &'a self,
+        f: &mut F,
```

Comment:
Is this a double reference? Could just `f: impl FnMut(LexicallyScopedDeclaration<'a>)` work?